dali/pipeline/operators/util/optical_flow.cc

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <dali/pipeline/data/views.h>
+#include <dali/pipeline/operators/util/optical_flow.h>
+
+namespace dali {
+
+
+DALI_SCHEMA(OpticalFlow)
+                .DocStr(R"code(Calculates the Optical Flow for sequence of images given as a input.
+ Mandatory input for the operator is a sequence of frames.
+ As as optional input, operator accepts external hints for OF calculation.
+The output format of this operator matches output format of OF driver API.
+CPU version of this Operator is no-op.)code")
+                .NumInput(1, 2)
+                .NumOutput(1)
+                .AddOptionalArg(detail::kPresetArgName, R"code(Setting quality level of OF calculation.
+ 0.0f ... 1.0f, where 1.0f is best quality, lowest speed)code", .0f, false)
+                .AddOptionalArg(detail::kOutputFormatArgName,
+                                R"code(Setting grid size for output vector.)code", -1, false)
+                .AddOptionalArg(detail::kEnableHintsArgName,
+                                R"code(enabling/disabling temporal hints for sequences longer than 2 images)code",
+                                false, false);
+
+DALI_REGISTER_OPERATOR(OpticalFlow, OpticalFlow<CPUBackend>, CPU);
+
+DALI_REGISTER_OPERATOR(OpticalFlow, OpticalFlow<GPUBackend>, GPU);
+
+
+template<>
+void OpticalFlow<CPUBackend>::RunImpl(Workspace<CPUBackend> *ws, const int idx) {
+  const auto &input = ws->template Input<CPUBackend>(idx);
```

Comment:
```suggestion
  const auto &input = ws->Input<CPUBackend>(idx);
```
The template disambiguator is not needed in template specialization

Reply:
Done